app/models/order_cycle_shipping_method.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+class OrderCycleShippingMethod < ApplicationRecord
+  belongs_to :order_cycle
+  belongs_to :shipping_method, class_name: "Spree::ShippingMethod"
+
+  validate :shipping_method_belongs_to_order_cycle_distributor
+  validates :shipping_method, uniqueness: { scope: :order_cycle_id }
+
+  private
+
+  def shipping_method_belongs_to_order_cycle_distributor
+    return if order_cycle.nil? ||
+              shipping_method.nil? ||
+              shipping_method.distributors.where(id: order_cycle.distributor_ids).exists?
+
+    errors.add(:shipping_method, :must_belong_to_order_cycle_distributor)
+  end
+end
```

Comment:
I think that this model is very clever but after all the changes we made, I'm wondering if we still need it. The validation could move to the order cycle, I guess. The uniqueness can be enforced in the database. And I think that Rails can do the association via this table without a separate model.

I'm undecided on this. What do you think? Do we need it?

Reply:
Removing it sounds good, it seems to make things simpler again, see https://github.com/openfoodfoundation/openfoodnetwork/pull/9262/commits/98a3318dcbd1f9f4379d4966ded32d3af59fc68e

Now, instead of the validation the `OrderCycleForm` just ignores invalid shipping methods which shouldn't be attached, which is the same way invalid schedules are [handled](https://github.com/openfoodfoundation/openfoodnetwork/blob/master/spec/services/order_cycle_form_spec.rb#L102).